packages/ag-charts-community/src/chart/axis/categoryAxis.ts

```diff
@@ -1,16 +1,19 @@
 import type { ModuleContext } from '../../module/moduleContext';
 import { BandScale } from '../../scale/bandScale';
+import type { OrdinalTimeScale } from '../../scale/ordinalTimeScale';
 import { RATIO, Validate } from '../../util/validation';
 import { CartesianAxis } from './cartesianAxis';
 
-export class CategoryAxis extends CartesianAxis<BandScale<string | object>> {
-    static readonly className = 'CategoryAxis';
-    static readonly type = 'category' as const;
+export class CategoryAxis<
+    S extends BandScale<string | object> | OrdinalTimeScale = BandScale<string | object>,
+> extends CartesianAxis<S> {
+    static readonly className: 'CategoryAxis' | 'OrdinalTimeAxis' = 'CategoryAxis';
```

Comment:
Nit: Make this type `string` if possible, and it avoids needing to declare all sub-classes?